SpringKonstant/src/org/usfirst/frc6357/robotcode/subsystems/TFMini.java

```diff
@@ -148,91 +160,80 @@ private void initialize()
      * Runs a loop in a Thread that will read the 9 byte packets and set each
      * byte to the correct piece of data for use if the data is valid.
      */
```

Comment:
This function is now "run" which is the function invoked when .start() is called on a Thread object. When using the old method, we were creating a lambda function (the "() -> {}" is the java lambda syntax) to hand to the new Thread. Both Thread itself and a lambda implement IRunnable. IRunnable defines the run function, which the Thread .start() function calls.

By defining run, we are doing exactly the same thing as before, except we define the runnable function as a member of the class rather than a lambda. This allows us to use the singleton pattern on the same object. We can still call start multiple times, but we would get an IllegalThreadStateException if the thread is already running. We can still stop the thread by setting startRead to false in the stop function and waiting for the thread to join the caller.